Refactor the builder

ilija-lazoroski opened the pull request.

ilija-lazoroski committed:
Make load_package_names private

ilija-lazoroski committed:
Move long list of CLI options into separate module

ilija-lazoroski committed:
Rename build_{dir, dir_path} for consistency

ilija-lazoroski committed:
Rename plugin_{path, dir_path} for consistency

ilija-lazoroski committed:
Rename source_{dirname|dir, dir_name} for consistency

ilija-lazoroski committed:
Rename {check_if, should_use}_common_vendor_dir_{possible, }

ilija-lazoroski committed:
Rename dir to log_directory in logging

ilija-lazoroski committed:
Rename _run_{container_with_plugin_dir, command_in_docker_container}

ilija-lazoroski committed:
Rename vendor_dirs to vendor_dir_generation

ilija-lazoroski committed:
Move PlatformDependencyPackagingMethod to separate file

ilija-lazoroski committed:
Refactor AgentPluginBuildOptions

ilija-lazoroski committed:
Rename build_{ ,agent}_plugin

mssalvatore commented on README.md:
```diff
@@ -33,7 +33,7 @@ where:
         -d/--dist-dir-path: The path where resulting Agent Plugin archive will be stored.
         Default: <current_working_directory>/dist
 
-        -s/--source-dir: The name of the source directory.
+        -s/--source-dir-name: The name of the source directory.
```
We have `--dist-dir-path` and `--source-dir-name`. Do these take different kinds of input? If not, we should shorten them to `--dist-dir` and `--source-dir`.

ilija-lazoroski replied:
Yep they do. `--dist-dir-path` gets path and `--source-dir-name` a string

mssalvatore commented on tests/test_agent_plugin_build_options.py:
Add instructions on running the test suite to the README.md.

mssalvatore commented on tests/test_agent_plugin_build_options.py:
I realize this is a draft PR at the moment. This is where we stand ATM: 
![image](https://github.com/user-attachments/assets/f7816a9c-38f9-4b5e-b2af-c7c825584d50)

ilija-lazoroski replied:
Yep. Until now only `AgentPluginBuildOptions` has been tested so that kind of explains it.

ilija-lazoroski committed:
Add note in README for running unit tests

ilija-lazoroski committed:
Add tests for setting up logging

ilija-lazoroski committed:
Refactor vendor_dir_generation

ilija-lazoroski committed:
Refactor build_agent_plugin into multiple packages

ilija-lazoroski committed:
Reorder methods in vendor_dir_generation

ilija-lazoroski committed:
Add tests for generate_vendor_directories

ilija-lazoroski committed:
Add unit tests for plugin_manifest

ilija-lazoroski committed:
Add tests for plugin_schema_generation

ilija-lazoroski committed:
Add unit tests for plugin_archive_generation

ilija-lazoroski committed:
Add integration tests for plugin_archive_generation

ilija-lazoroski committed:
Rename build_agent_plugin{ , _archive}

ilija-lazoroski committed:
Add tests for CustomArgumentsFormatter

I know I am testing private method but it is our code
so we need to take care of it

ilija-lazoroski committed:
Reduce AgentPluginManifest duplication in UTs

ilija-lazoroski committed:
Reduce AgentPluginBuildOptions duplication in UTs

ilija-lazoroski committed:
Add missing type hints to UTs

ilija-lazoroski committed:
Fix autodetect bug where only one OS is supported

ilija-lazoroski committed:
Add note for skipping integration tests in README

ilija-lazoroski committed:
Add fixed entry for AUTODETECT bug fix

mssalvatore committed:
Bump version to v0.6.0
